core/src/main/java/com/sequenceiq/cloudbreak/conclusion/ConclusionChecker.java

```diff
@@ -25,25 +27,28 @@ public ConclusionChecker(List<ConclusionStep> conclusionSteps) {
 
     public ConclusionResult doCheck(Long resourceId) {
         LOGGER.info("Conclusion checker started, steps: {}, resourceId: {}", conclusionSteps, resourceId);
-        List<String> conclusions = new ArrayList<>();
+        List<Conclusion> conclusions = new ArrayList<>();
         try {
             while (hasMoreSteps() && doNext()) {
                 ConclusionStep conclusionStep = conclusionSteps.get(actualStep++);
                 LOGGER.debug("Conclusion step: {}", conclusionStep);
 
-                ConclusionStepResult stepResult = conclusionStep.check(resourceId);
-                LOGGER.debug("Conclusion step {}, conclusion: {}", stepResult.isStepFailed() ? "failed" : "succeeded", stepResult.getConclusion());
-                if (stepResult.getConclusion() != null) {
-                    conclusions.add(stepResult.getConclusion());
+                Conclusion conclusion = measure(() -> conclusionStep.check(resourceId),
+                        LOGGER, "Conclusion step finished in {} ms for resourceId {}, step {}", resourceId, conclusionStep.getClass().getSimpleName());
+                if (conclusion.isStepFailed()) {
+                    LOGGER.debug("Conclusion step failed, conclusion: {}, details: {}", conclusion.getConclusion(), conclusion.getDetails());
```

Comment:
What does step failed mean in this context? 
Is it that we were not able to execute the check due too unexpected error or the result we were expecting was marking some form of error?
So in other words did it encounter a failure during it's execution, or properly detected an error?

Reply:
The latter, it means the result we were expecting was marking some form of error.